### PR TITLE
Set up pre-commit hooks and added configuration files

### PR DESCRIPTION
**Pull request by Al-an-21**
PR introduces pre-commit hooks and updates the layer.conf file to ensure compatibility with the specified Yocto series.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      - id: codespell
+
+  - repo: https://github.com/adrienverge/yamllint
+    rev: v1.35.1
+    hooks:
+      - id: yamllint
+        args: [-c=.yamllint]
+
+  - repo: https://github.com/3mdeb/hooks
+    rev: v0.1.2
```
Comment (PLangowski) [inline .pre-commit-config.yaml:45]: I just pushed `v0.1.3` because I forgot to do that after recent changes. We should pull that release.

**Review by PLangowski — CHANGES_REQUESTED** [review]
The `pre-commit-init` directory shouldn't be here. You should have cloned https://github.com/3mdeb/hooks and executed `pre-commit-init.py` with arguments. The directory is used to generate pre-commit configs and shouldn't be added to the repositories.

**Comment by Al-an-21** [comment]
@PLangowski So in this case, I should redo the process or will removing the `pre-commit-init` directory be enough?

**Comment by PLangowski** [comment]
@Al-an-21 The config looks fine to me. Did you copy the files from the `3mdeb/hooks` repo?

**Comment by Al-an-21** [comment]
> @Al-an-21 The config looks fine to me. Did you copy the files from the `3mdeb/hooks` repo?

@PLangowski Yes

**Comment by PLangowski** [comment]
@Al-an-21 Ok, I just tried it myself and there were no problems with running the pre-commits. So if you delete the directory and bump the `namespell` revision, I will approve the changes

File: .pre-commit-config.yaml (hunk 1 above)
Comment (artur-rs) [reply .pre-commit-config.yaml]: 0.1.3 is already added

**Comment by artur-rs** [comment]
@Al-an-21 resolved the therads, but as you can see, pre-commit failed: https://results.pre-commit.ci/run/github/884698451/1731078179.50Y0Yv9MRdSin6rqHVUy-A
Please check whats wrong and try to fix it.

**Review by macpijan — APPROVED** [review]

**Pull request merged**